packages/wrangler/src/metrics/metrics-dispatcher.ts

```diff
@@ -1,80 +1,156 @@
+import chalk from "chalk";
 import { fetch } from "undici";
-import { version as wranglerVersion } from "../../package.json";
+import isInteractive from "../is-interactive";
 import { logger } from "../logger";
-import { getMetricsConfig } from "./metrics-config";
+import { CI } from "./../is-ci";
+import {
+	getConfigFileType,
+	getNodeVersion,
+	getOS,
+	getOSVersion,
+	getPackageManager,
+	getPlatform,
+	getWranglerVersion,
+} from "./helpers";
+import {
+	getMetricsConfig,
+	readMetricsConfig,
+	writeMetricsConfig,
+} from "./metrics-config";
 import type { MetricsConfigOptions } from "./metrics-config";
+import type { CommonEventProperties, Events } from "./types";
 
-// The SPARROW_SOURCE_KEY is provided at esbuild time as a `define` for production and beta
-// releases. Otherwise it is left undefined, which automatically disables metrics requests.
-declare const SPARROW_SOURCE_KEY: string;
 const SPARROW_URL = "https://sparrow.cloudflare.com";
 
-export async function getMetricsDispatcher(options: MetricsConfigOptions) {
+export function getMetricsDispatcher(options: MetricsConfigOptions) {
+	// The SPARROW_SOURCE_KEY will be provided at build time through esbuild's `define` option
+	// No events will be sent if the env `SPARROW_SOURCE_KEY` is not provided and the value will be set to an empty string instead.
+	const SPARROW_SOURCE_KEY = process.env.SPARROW_SOURCE_KEY ?? "";
+	const requests: Array<Promise<void>> = [];
+	const wranglerVersion = getWranglerVersion();
+	const amplitude_session_id = Date.now();
```

Comment:
This won't be stable across `getMetricsDispatcher()` calls. Is that a problem?

Reply:
that's okay -`getMetricsDispatcher()` is only called once for the command start/end events, which are the only events that actually use the amplitude session id. Later it would be good to consolidate the 'adhoc' events to use this session id too but thought it'd be better to leave that for when we can provide the dispatcher through `defineCommand`